pkg/lvm/lvm_util.go

```diff
@@ -254,6 +256,94 @@ func ResizeLVMVolume(vol *apis.LVMVolume, resizefs bool) error {
 	return err
 }
 
+// ResizeBTRFSVolume resizes the LVM volume as well as expand btrfs
+// Step1: Expanding LVM Volume (lvextend lvmvg -n <volume_name> -L size)
+// Step2: Expanding btrfs filesystem (btrfs filesystem resize max <mount_path>)
+func ResizeBTRFSVolume(vol *apis.LVMVolume, mountPath string) error {
+
+	desiredVolSize, err := strconv.ParseUint(vol.Spec.Capacity, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	curVolSize, err := getVolumeSize(vol)
+	if err != nil {
+		return err
+	}
+
+	// Handle a case where LVM is expanded but not the filesystem
+	if desiredVolSize > curVolSize {
+		err := ResizeLVMVolume(vol, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	args := []string{"filesystem", "resize", "max", mountPath}
+	cmd := exec.Command(BTRFS, args...)
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		klog.Errorf(
+			"lvm could not expand btrfs filesystem of volume %v cmd %v error: %s",
+			filepath.Join(vol.Spec.VolGroup, vol.Name),
+			args,
+			string(out),
+		)
+		return errors.Wrap(err, string(out))
+	}
+	return nil
+}
+
+// getVolumeSize will return current LVM volume size on bytes
+func getVolumeSize(vol *apis.LVMVolume) (uint64, error) {
```

Comment:
can we change it to `getLVSize()`, since we already have a `getVGSize()` function.

Reply:
Updated